src/parser/SpatialQuery.h

```diff
@@ -0,0 +1,57 @@
+// Copyright 2024, University of Freiburg
+// Chair of Algorithms and Data Structures
+// Authors: Christoph Ullinger <ullingec@informatik.uni-freiburg.de>
+
+#pragma once
+
+#include "parser/GraphPattern.h"
+#include "parser/MagicServiceQuery.h"
+
+namespace parsedQuery {
+
+class SpatialSearchException : public std::exception {
+  std::string message_;
+
+ public:
+  explicit SpatialSearchException(const std::string& message)
+      : message_(message) {}
+  const char* what() const noexcept override { return message_.data(); }
+};
+
+// Spatial Search feature via SERVICE. This struct holds intermediate or
+// incomplete configuration during the parsing process.
+struct SpatialQuery : MagicServiceQuery {
+  // Required after everything has been added: the left and right join
+  // variables.
+  std::optional<Variable> left_;
+  std::optional<Variable> right_;
+
+  // The spatial join task definition: maximum distance and number of results.
+  // One of both - or both - must be provided.
+  std::optional<size_t> maxDist_;
+  std::optional<size_t> maxResults_;
+
+  // Optional further arguments
+  std::optional<Variable> bindDist_;
+  std::optional<SpatialJoinAlgorithm> algo_;
```

Comment:
Document them (in particular the `bindDist` because it cannot fully be understood by the type.
`variableForDistance` is a better name. or `distanceVariable_` (I prefer the latter).